### PR TITLE
Fix wxStringOutputStream position when using external string

### PR DESCRIPTION
**Pull request by vadz**
See https://trac.wxwidgets.org/ticket/17985

### CODE REVIEW

File: src/common/sstream.cpp [hunk 1]
```diff
@@ -128,6 +128,21 @@ size_t wxStringInputStream::OnSysRead(void *buffer, size_t size)
...
+wxStringOutputStream::wxStringOutputStream(wxString *pString, wxMBConv& conv)
+    : m_conv(conv)
+#if wxUSE_UNICODE
+    , m_unconv(0)
+#endif // wxUSE_UNICODE
+{
+    m_str = pString ? pString : &m_strInternal;
+
+#if wxUSE_UNICODE
+    m_pos = m_conv.FromWChar(NULL, 0, m_str->wc_str(), m_str->length());
```
Comment (paulcor) [inline src/common/sstream.cpp]: Wouldn't `m_pos = m_str->mb_str(conv).length();` be more efficient for the `wxUSE_UNICODE_UTF8` case (and a little bit simpler)?

File: tests/streams/sstream.cpp [hunk 2]
```diff
@@ -119,3 +119,32 @@ void strStream::Output_Check()
...
+TEST_CASE("wxStringOutputStream::Tell", "[stream]")
+{
+    wxStringOutputStream ss;
+    CHECK( ss.TellO() == 0 );
+
+    const char* const s = "Hello world";
+
+    ss.Write(s, strlen(s));
+    const wxFileOffset pos = ss.TellO();
+    CHECK( pos != 0 );
```
Comment (paulcor) [inline tests/streams/sstream.cpp]: `pos != 0` is a pretty low bar... I get that you're using the result in the next test, but shouldn't the result really be `strlen(s)`?

Comment (vadz) [reply tests/streams/sstream.cpp]: Yes, thanks, I initially wanted to reuse the same code for both the UTF-8 and UTF-16 checks and then the position would differ, but with the way it ended up being written, it should indeed be just checked directly, I'll update this.

File: src/common/sstream.cpp (hunk 1 above)
Comment (vadz) [reply src/common/sstream.cpp]: I think using `FromWChar(NULL)` is faster as it doesn't actually perform the conversion, but I've added an even faster path for the `wxUSE_UNICODE_UTF8` case now.

**Pull request merged**